src/shipchain_common/test_utils/httpretty_asserter.py

```diff
@@ -73,9 +73,21 @@ def _assert_call_in_list(cls, call, assertion):
             assert assertion['query'] == call['query'], \
                 f'Error: query mismatch, desired `{assertion["query"]}` returned `{call["query"]}`.'
 
+    @classmethod
+    def reset_calls(cls):
+        # If calling cls.reset(), latest_requests are unable to build.
+        # However, if calls were mocked in previous tests and were not asserted, then it can cause issues when
+        # asserting calls in a specific test.
+
+        for index, call in enumerate(cls.latest_requests):
+            if call is None:
+                continue
+            cls.latest_requests[index] = None
```

Comment:
If all you need to do is clear out the array, can you not just do `cls.latest_requests = []`? Or if for some reason you need to have an array with n number of `None`s, `cls.latest_requests = [None for call in cls.latest_requests]`

Reply:
Due to how http-pretty works, setting `cls.latest_requests = []` causes it to be unable to parse through calls in later iterations. It makes sense to have it as a one line call however.

Reply:
Turns out you have to be very specific in replacing each item in the latest requests with a `None`, just setting it to a list of `None` of the same length causes it to be unable to add to it again.